go/libraries/utils/argparser/parser.go

```diff
@@ -49,15 +49,31 @@ func ValidatorFromStrList(paramName string, validStrList []string) ValidationFun
 }
 
 type ArgParser struct {
+	Name              string
+	MaxArgs           int
+	TooManyArgsError  func(receivedArgs []string) error
 	Supported         []*Option
 	nameOrAbbrevToOpt map[string]*Option
 	ArgListHelp       [][2]string
 }
 
-func NewArgParser() *ArgParser {
+func NewArgParserWithVariableArgs(name string) *ArgParser {
```

Comment:
(minor) Give public/exported functions godocs (as well as any non-simple private functions). This makes a big difference to other people reading your code. For example, from just looking at the call site, it's not obvious why `name` is required here. Add some godocs for these new public functions that explains what they do, their params, return vals, etc. Doesn't need to be anything fancy, but think about what will be most helpful to future code readers who don't know these implementations.

Reply:
Fixed with https://github.com/dolthub/dolt/pull/5771